File: Codeforces/Practice/1600/shruti.py
t = int(input())
for _ in range(t):
    n = int(input())
    l = list(map(int, input().split()))
    mx = max(l)
    i=0
    ans = 0
    while(i < n):
        cnt = 0
        while(i < n and l[i] == mx):
            cnt += 1
            i += 1
        i += 1
        ans = max(ans, cnt)
    print(ans)

File: Codeforces/Practice/1600/shruti.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int t; cin>>t;
    while(t--){
        int n; cin>>n;
        string s; cin>>s;
        int i = 0;
        int ans = 1;
        while(i < n){
            int cnt = 0;
            char ch = s[i];
            while(i < n && s[i] == ch){
                i++;
                cnt++;
            }
            ans *= cnt;
            ans %= 10000007;
        }
        cout<<ans<<endl;
    }
    return 0;
}